uktrade/data-hub-frontend
# Add company to user list.

## Description of change
Trello - https://trello.com/c/kHaS9RTG/838-companies-list-add-and-remove-company

Users need to be able to "add" and "remove" companies to their own personal list which will soon be surfaced on the home page of Data Hub. 

## Test instructions

Note: You will need to set up a feature flag when running Leeloo locally, the flag is called `add_company_to_list`

1. Go to companies
2. Choose company
3. Click"add to my list" > You should see a success message
4. Click "remove from my list" > you should see a "add to my list" button re-appear
5. Archive the company and you should still see a "remove from my list" button if its in your list but not a "add to my list" button if it isn't.


 
## Screenshots
### Before
![Screenshot 2019-07-31 at 10 30 39](https://user-images.githubusercontent.com/10154302/62201031-7ca5c280-b37e-11e9-9739-d58e90b1acc1.png)


### After 
#### Company not added
![Screenshot 2019-07-31 at 10 31 07](https://user-images.githubusercontent.com/10154302/62201230-e8882b00-b37e-11e9-9006-5accdc896714.png)

#### Company added
![Screenshot 2019-07-31 at 10 31 16](https://user-images.githubusercontent.com/10154302/62201262-f63db080-b37e-11e9-9913-d4447c90eebe.png)

#### Company removed
![Screenshot 2019-07-31 at 10 31 24](https://user-images.githubusercontent.com/10154302/62201283-fe95eb80-b37e-11e9-9277-502d09c04537.png)

#### Company added but archived
![Screenshot 2019-07-31 at 10 31 51](https://user-images.githubusercontent.com/10154302/62201304-0b1a4400-b37f-11e9-81e4-ba4f6396d74e.png)


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied. 
https://github.com/uktrade/data-hub-frontend/blob/develop/docs/Code%20review%20guidelines.md"
